kalibro_client/processor/repository.py

```diff
@@ -1,6 +1,132 @@
-    pass
+class Repository(attributes_class_constructor('RepositoryAttr',
+                                              (('name', None),
+                                               ('description', None),
+                                               ('license', None),
+                                               ('scm_type', None),
+                                               ('address', None),
+                                               ('code_directory', None),
+                                               ('branch', None))), Base):
+    def __init__(self, period=None, project_id=None,
+                 kalibro_configuration_id=None, *init_args, **init_kwargs):
+        super(Repository, self).__init__(*init_args, **init_kwargs)
+        self.period = period
+        self.project_id = project_id
+        self.kalibro_configuration_id = kalibro_configuration_id
+
+    @property
+    def period(self):
+        return self._period
+
+    @period.setter
+    def period(self, value):
+        if value is not None:
+            value = int(value)
+
+        self._period = value
+
+    @property
+    def project_id(self):
+        return self._project_id
+
+    @project_id.setter
+    def project_id(self, value):
+        if value is not None:
+            value = int(value)
+
+        self._project_id = value
+
+    @property
+    def kalibro_configuration_id(self):
+        return self._kalibro_configuration_id
+
+    @kalibro_configuration_id.setter
+    def kalibro_configuration_id(self, value):
+        if value is not None:
+            value = int(value)
+
+        self._kalibro_configuration_id = value
+
+    def process(self):
+        return self.request(action='/process', params={'id': self.id},
+                            method='get')
+
+    def cancel_processing_of_a_repository(self):
+        return self.request(action=':id/cancel_process',
+                            params={'id': self.id},
+                            method='get')
+
+    def processing(self):
+        if self.has_ready_processing():
+            return self.last_ready_processing()
+        return self.last_processing()
+
+    def processing_with_date(self, date):
+        date = dateutil.parser.parse(date) \
+            if isinstance(date, (str, unicode)) else date
+        if self.has_processing_after(date):
+            return self.first_processing_after(date)
+        elif self.has_processing_before(date):
+            return self.last_processing_before(date)
+        else:
+            return None
+
+    def has_processing(self):
+        return self.request(':id/has_processing',
+                            params={'id': self.id},
+                            method='get')['has_processing']
```

Comment:
All the `processing` methods are returning a hash right now? Shouldn't they return a `Processing` with the method `response_to_objects_array`?